econml/ortho_forest.py

```diff
@@ -758,6 +763,35 @@ def parameter_estimator_func(Y, T,
         # If any of the values in the parameter estimate is nan, return None
         return param_estimate
 
+    @staticmethod
+    def second_stage_parameter_estimator_gen(lambda_reg):
+        """ For the second stage parameter estimation we add a local linear correction. So
+        we fit a local linear function as opposed to a local constant function. We also penalize
+        the linear part to reduce variance. """
+        def parameter_estimator_func(Y, T, X,
+                                     nuisance_estimates,
+                                     sample_weight=None):
+            """Calculate the parameter of interest for points given by (Y, T) and corresponding nuisance estimates."""
+            # Compute partial moments
+            pointwise_params = DiscreteTreatmentOrthoForest._partial_moments(Y, T, nuisance_estimates)
+            X_aug = PolynomialFeatures(degree=1, include_bias=True).fit_transform(X)
+            # Compute coefficient by OLS on residuals
+            if sample_weight is not None:
+                weighted_X_aug = sample_weight.reshape(-1, 1) * X_aug
+            else:
+                weighted_X_aug = X_aug / X_aug.shape[0]
+            # ell_2 regularization
+            diagonal = np.ones(X_aug.shape[1])
+            diagonal[0] = 0
+            reg = lambda_reg * np.diag(diagonal)
+            # Ridge regression estimate
+            param_estimate = np.matmul(np.linalg.pinv(np.matmul(weighted_X_aug.T, X_aug) + reg),
```

Comment:
Why `pinv` instead of `inv` if we're doing ridge?  Shouldn't the regularization force the matrix to be invertible?

Also, when numerically computing (A^-1* b) you can avoid an explicit matrix inversion by using `np.linalg.solve(A,b)` instead of `np.matmul(np.linalg.inv(A), B)` (or `lstsq` in the `pinv` case), which should be more efficient and accurate.

Reply:
just in case the user specifies lambda_reg=0.

Reply:
Makes sense.  I still think that you should use
```suggestion
            param_estimate = np.linalg.lstsq(np.matmul(weighted_X_aug.T, X_aug) + reg,
```

(replacing the outer `np.matmul(np.linalg.pinv(...), ...)` with `np.linalg.lstsq`) for efficiency/numerical stability.

Reply:
Done